### PR TITLE
std.debug: Fix defaultPanic() uefi build

### PR DESCRIPTION
**Pull request by linusg**
See commit message. Confirmed this builds but not tested as either my local QEMU setup broke or there is a larger regression somewhere else (I can't get it to boot).

### CODE REVIEW

File: lib/std/debug.zig [hunk 1]
```diff
@@ -507,7 +507,7 @@ pub fn defaultPanic(
                 // ExitData buffer must be allocated using boot_services.allocatePool (spec: page 220)
                 const exit_data: []u16 = uefi.raw_pool_allocator.alloc(u16, exit_msg.len + 1) catch @trap();
                 @memcpy(exit_data, exit_msg[0..exit_data.len]); // Includes null terminator.
-                _ = bs.exit(uefi.handle, .Aborted, exit_msg.len + 1, exit_data);
+                _ = bs.exit(uefi.handle, .Aborted, exit_msg.len + 1, @ptrCast(exit_data));
```
Comment (rohlem) [inline lib/std/debug.zig:510]: Probably doesn't matter much, but I think `exit_data.ptr` would be more readable here.
For `@ptrCast` a reader has to look up the destination type (`?*const anyopaque)` to understand what it does / is used for here.

Comment (linusg) [reply lib/std/debug.zig]: sure, updated

**Comment by linusg** [comment]
> either my local QEMU setup broke or there is a larger regression somewhere else

Never mind, I mixed up x86-uefi-gnu and x86_64-uefi-gnu. Confirmed this works as intended:

image

**Review by alexrp — APPROVED** [review]

**Comment by andrewrk** [comment]
@alexrp For a 1 commit PR please do a rebase & merge because it avoids an unnecessary "merge commit" in master branch.

**Pull request merged**